docs/source/index.rst

```diff
@@ -0,0 +1,136 @@
+The DeepChem Project
```

Comment:
I modified two things.

- Make index.rst simpler (I think the simpler the index pages is, the more people reads)
  - Move license to `development_guide` section
  - Merge `Getting Involved` to ` About US` section
- Update the outline in the bottom of this file